pyOCD/gdbserver/gdbserver.py

```diff
@@ -1280,21 +1280,15 @@ def getThreadsXML(self):
         if not self.is_threading_enabled():
             t = SubElement(root, 'thread', id="1", core=str(self.core))
             if self.is_target_in_reset():
-                t.text = "Reset"
+                t.set('name', "Reset")
             else:
-                t.text = "Thread mode"
+                t.set('name', "Thread mode")
```

Comment:
Eventually we need to properly report all core states when there isn't a thread provider, such as handler mode, sleeping, lockup. (Just thinking out loud. 😄)

Reply:
Yes, and I realise this change is wrong - that isn't the thread name, it's the extra info, so should stay in the text. Doesn't make sense to have a thread name for the single thread.

I'll take out the "Thread mode" for now, as it's currently wrong, and leave the "Reset" as description.

A follow-up to add state from IPSR as annotation in place of "Thread mode" (Thread/HardFault/SVCall/Interrupt 23...) would make sense.

If the logic of #430 was followed through to give 2 threads for non-OS, then they would have the names "Main stack" and "Process stack". They could both retain the same IPSR-based annotation (although for Process it would always be Thread).